_source/_posts/2022-06-01-cloud-native-java-microservices-with-istio.md

```diff
@@ -0,0 +1,604 @@
+```
+
+Once the deployments are done, we must wait for the pods to be in **RUNNING** status. Useful links will be printed on the terminal; make a note of them.
+
+{% img blog/cloud-native-java-microservices-with-istio/jhipster-deployment.png alt:"GKE cluster with application pods" width:"900" %}{: .center-image }
+
+You can now access the application at the given `http://store.jhipster.<istio-ingress-gateway-external-ip>.nip.io` URI and log in with the default credentials.
+
+{% img blog/cloud-native-java-microservices-with-istio/jh-store-app.png alt:"Store gateway application" width:"900" %}{: .center-image }
+
+Currently the JHipster OIDC setup does not work with Istio and there is an [open issue](https://github.com/jhipster/generator-jhipster/issues/17384) in JHipster issue tracker for this. Alternative solutions would be to use an [external authorization server](https://istio.io/latest/blog/2021/better-external-authz/) with something like [Open Policy Agent](https://www.openpolicyagent.org/). We will cover this in a later blog post.
+
+### Monitoring and observability
+
+Since we deployed tools for observability, let's see what we have.
+
+**Grafana**
+
+First up are Grafana and Prometheus for metrics and dashboards. Click the URI for Grafana from the previous deployment step. Click **General** at the top left corner and click the **istio** folder. You should see multiple pre-configured dashboards. You can monitor the performance of the workloads and the istio system itself here. You can also create your own dashboards if you like. Prometheus provides the data visualized on Grafana.
+
+{% img blog/cloud-native-java-microservices-with-istio/grafana.png alt:"Grafana dashboard" width:"900" %}{: .center-image }
+
+**Kiali**
+
+Kiali is a management console for Istio service mesh, and it provides a web interface for visualizing the network topology of your service mesh. You can use it to explore the network topology of your cluster and see the network traffic flowing through it. Click **Graph** on the left side menu to see the network topology.
+
+{% img blog/cloud-native-java-microservices-with-istio/kiali.png alt:"Kiali dashboard" width:"900" %}{: .center-image }
+
+**Zipkin**
+
+Zipkin is a distributed tracing solution for distributed systems. It is a tool for capturing distributed traces and providing a centralized view of the traces. This is essential for a microservice setup where a request could span multiple services, and debugging would require tracing them. Click **RUN QUERY** on the home screen to fetch recent traces, and click **SHOW** on one of them.
+
+{% img blog/cloud-native-java-microservices-with-istio/zipkin.png alt:"Zipkin dashboard" width:"900" %}{: .center-image }
+
+### Cleanup
+
+Once you are done with experiments, make sure to delete the cluster you created so that you don't end up with a big bill from Google. You can delete the cluster from the Google Cloud Console GUI or via the command line using the below command.
```

Comment:
```suggestion
Once you are done with experiments, make sure to delete the cluster you created so that you don't end up with a big bill from Google. You can delete the cluster from the Google Cloud Console GUI or via the command line using the following command.
```